src/scheduler/resource.c

```diff
@@ -872,12 +872,12 @@ collect_resources_from_requests(resource_resv **resresv_arr)
 		 * execselect: select created from the exec_vnode.  This is likely to
 		 * be a subset of resources from schedselect + 'vnode'.  It is possible
 		 * that a job was run with qrun -H (res=val) where res is not part of
-		 * the schedselect.  The exec_vnode is taken directly from the -H  argument
+		 * the schedselect.  The exec_vnode is taken directly from the -H argument
 		 * The qrun -H case is why we need to do this check.
 		 */
 		if (r->execselect != NULL && r->execselect->defs != NULL) {
-			if ((r->is_job && r->job != NULL && in_runnable_state(r)) ||
-				(r->is_resv && r->resv != NULL && r->resv->resv_state == RESV_BEING_ALTERED)) {
+			if ((r->job != NULL && in_runnable_state(r)) ||
+				(r->resv != NULL && (r->resv->resv_state == RESV_BEING_ALTERED || r->resv->resv_substate == RESV_DEGRADED))) {
```

Comment:
do you not need is_job or is_resv?

Reply:
Yeah, I don't necessary need it, but unless we refactor is_job and is_resv out out of the code, I guess they should be used.  I'll remove them if you want.